src/components/virtualRepeat/virtual-repeater.js

```diff
@@ -497,7 +498,14 @@ VirtualRepeatController.prototype.readItemSize_ = function() {
 
   this.items = this.repeatListExpression(this.$scope);
   this.parentNode = this.$element[0].parentNode;
+
   var block = this.getBlock_(0);
+
+  // Make sure the block is fully constructed
+  if (!this.$rootScope.$$phase) {
+    this.$rootScope.$apply();
+  }
```

Comment:
This strikes me as the wrong approach. Forcing a digest of the block's scope should be sufficient.